src/index.js

```diff
@@ -127,14 +129,20 @@ async function main(req, context) {
       log,
     });
 
-    const useCDN = true;
+    if (useLastModified) {
+      const metadata = await contentStorage.metadata(key);
+      if (metadata) {
+        options.lastModified = metadata['last-modified'];
+      }
+    }
+
     const res = await contentProxy({
       owner, repo, ref, path, mp, log, options, resolver, useCDN,
     });
     if (!res.ok) {
```

Comment:
how does this work. does a 304 cause a `!res.ok` ?

Reply:
exactly: `res.ok` is defined as `(statusCode >= 200 && statusCode < 300)`, so it just forwards the `Not Modified` response